test/stress/cpp/microbench.cpp

```diff
@@ -39,9 +41,9 @@ new_sized_delete(void) {
 
 static void
 malloc_sdallocx(void) {
-	void *p = malloc(1);
-	expect_ptr_not_null(p, "Unexpected malloc failure");
-        sdallocx(p, 1, 0);
+	auto sdallocfn = [&](char *p){sdallocx(p, 1, 0);};
+	std::unique_ptr<char, decltype(sdallocfn)> p {(char*)malloc(1), sdallocfn};
+	expect_ptr_not_null(p.get(), "Unexpected malloc failure");
```

Comment:
This is not buying anything but wanted to play with c++ memory wrappers a bit :)

Reply:
Haha understood. For now I feel we still want the jemalloc side to be in explicit control. But it would be interesting to verify that the wrappers add 0 cost.

Closing the PR but as always thanks for sharing the thoughts!